src/Controller/ParticipantDetailsController.php

```diff
@@ -305,6 +305,10 @@ public function participantIncentiveAction(
         if ($incentive && $incentive->getSite() !== $siteService->getSiteId()) {
             throw $this->createAccessDeniedException();
         }
+        if ($incentive && $incentive->getRecipient() == Incentive::PEDIATRIC_GUARDIAN) {
+            $incentive->setRecipient($incentive->getRelatedParticipantRecipient());
+        }
+        $incentive->setRecipient($incentive->getRelatedParticipantRecipient());
```

Comment:
If the intention is to set the recipient of $incentive to the related participant recipient only when the if condition is met, then the second setRecipient call should probably be removed.